pkg/kubectl/cmd/annotate.go

```diff
@@ -193,7 +204,11 @@ func (o AnnotateOptions) RunAnnotate() error {
 			return err
 		}
 
-		name, namespace, obj := info.Name, info.Namespace, info.Object
+		obj, err := resource.AsVersionedObject([]*resource.Info{info}, false, o.outputVersion.String())
```

Comment:
> I've used here the pattern we have in our env command specifically.



> Yeah, we had a long discussion about this in our env command.

This would seem to be a discussion I missed.  How can this be right?  I don't specify an output version.  Locally I have v1 and v2, v2 has priority.  v2 gets chosen, but v1 is the server-side level.  The patch is build using v2, which has labels under the tag `anything else`.  That patch gets sent to the v1 API and nothing happens.